driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -431,7 +439,7 @@ CloseFuture removePool(Host host) {
         }
 
         // Wait pool creation before removing, so we don't lose connectivity
-        ListenableFuture<?> allPoolsCreatedFuture = Futures.successfulAsList(poolCreatedFutures);
+        ListenableFuture<?> allPoolsCreatedFuture = Futures.allAsList(poolCreatedFutures);
```

Comment:
I have noticed that `void updateCreatedPools(Host h)` catches `InterruptedException` and `ExecutionException` and [does things](https://github.com/datastax/java-driver/pull/831/files#diff-8ccd18d8cb39074bf83554358d002ceeR476) if those exceptions were to be thrown. And I wonder if now that we use `allAsList()` it is something to be worried about in this method, too.